diff --git a/testing/dqlite/APKBUILD b/testing/dqlite/APKBUILD
index 7a77e2c8f7..d8a9b7468d 100644
--- a/testing/dqlite/APKBUILD
+++ b/testing/dqlite/APKBUILD
@@ -8,7 +8,7 @@ url="https://dqlite.io/"
 arch="all"
 license="Apache"
 depends="raft"
-makedepends="libuv-dev sqlite-dev sqlite-replication-dev
+makedepends="libuv-dev sqlite-dev
 raft-dev autoconf automake libtool"
 subpackages="$pkgname-static $pkgname-dev $pkgname-doc"
 source="$pkgname-$pkgver.tar.gz::https://github.com/canonical/dqlite/archive/v$pkgver.tar.gz"
diff --git a/testing/lxd/APKBUILD b/testing/lxd/APKBUILD
index c877ac920a..3c817c73fc 100644
--- a/testing/lxd/APKBUILD
+++ b/testing/lxd/APKBUILD
@@ -15,7 +15,6 @@ depends="
 	squashfs-tools
 	rsync
 	dqlite
-	sqlite-replication
 	shadow-uidmap
 	lxc
 	ip6tables
@@ -47,7 +46,6 @@ makedepends="
 	dqlite-dev
 	dqlite-static
 	lz4-dev
-	sqlite-replication-dev
 	sqlite-dev
 	sqlite-static
 	raft-dev
@@ -79,8 +77,7 @@ _project="github.com/lxc/lxd"
 build() {
 	export GOPATH="$builddir/_dist"
 	export GOFLAGS="-buildmode=pie -trimpath"
-	export CGO_CFLAGS="$CFLAGS -I/usr/include/sqlite-replication"
-	export CGO_LDFLAGS="-lintl $LDFLAGS -L/usr/lib/sqlite-replication -Wl,-R/usr/lib/sqlite-replication"
+	export CGO_LDFLAGS="-lintl $LDFLAGS"
 	export CGO_LDFLAGS_ALLOW="(-Wl,-wrap,pthread_create)|(-Wl,-z,now)"
 	export GO111MODULE=off
 
